app/scripts/controllers/preferences.js

```diff
@@ -37,6 +37,10 @@ export default class PreferencesController {
       // set to true means the dynamic list from the API is being used
       // set to false will be using the static list from contract-metadata
       useTokenDetection: false,
+      advancedGasFee: {
+        maxBaseFee: '',
+        priorityFee: '',
+      },
```

Comment:
`undefined` as default may be useful to know that user has no custom setting, WDYT ?

Reply:
I have updated the values to undefined now, it seems to be working.

Reply:
if user has not set default we can check as

1. `advancedGasFee` === undefined or
2. `advancedGasFee.maxBaseFee  == undefined && advancedGasFee.priorityFee  == undefined`

I am thinking what will be best way to check absence of default here.

Reply:
I would prefer to go with 2, it might be an extra check but I think we can reduce the chances of breaking the code if we miss checking for undefined for advancedGasFee anywhere in the future.

Reply:
I am not sure this is a proper use of `undefined`. I feel that `undefined` should only be used internally by JavaScript to indicate that a property of an object doesn't exist. What do we feel about either leaving these properties out entirely and making `advancedGasFee` an empty object (making them *de facto* `undefined`), or (and this might make more sense from a type perspective) setting them to `null`?

Reply:
Setting `advancedGasFee` to null sounds good to me.

Reply:
updated `advancedGasFee` to default to null